### PR TITLE
Issue 148: Update CRD version to v1

### LINKED ISSUE #148: Add apiVersion check to the CRD
### Description
The apiextensions.k8s.io/v1beta1 apiVersion used in the CustomResourceDefinition was removed in k8s v1.19.0

### Importance
The issue is a blocker to install the chart properly on k8s clusters with version v1.19.0

### Location
charts/bookkeeper-operator/templates/bookkeeper.pravega.io_bookkeeperclusters_crd.yaml

### Suggestions for an improvement
The issue can be fixed by checking the apiVersion with .Capabilities.APIVersions.Has

### PR DESCRIPTION
**Pull request by anishakj**
Signed-off-by: anishakj <anisha.kj@dell.com>

### Change log description

Update CRD version to `v1`
### Purpose of the change

 Fixes #148
### What the code does

Update CRD version, ClusterRoleBinding, ValidatingWebhookConfiguration. With the updated  controller run time , config map updation and finalizer updation was failing in unit tests,  Made changes to find the resource version of existing cluster and configmap and updated the resource version before doing update. Also passed ObjectType  field from  unit tests 

### How to verify it

Verified that able to install bookkeeper operator and bookkeeper cluster

### CODE REVIEW

**Comment by anishakj** [comment]
End to  end tests are failing because of a bug in `operator-sdk` (https://github.com/operator-framework/operator-sdk/pull/3121) Needs to update `operator-sdk` version.

File: pkg/controller/bookkeepercluster/bookkeepercluster_controller.go [hunk 1]
```diff
@@ -13,12 +13,13 @@ package bookkeepercluster
...
 	"context"
 	"fmt"
-	"github.com/pravega/bookkeeper-operator/pkg/controller/config"
 	"reflect"
 	"sort"
 	"strings"
 	"time"
 
+	"github.com/pravega/bookkeeper-operator/pkg/controller/config"
+
```
Comment (nishant-yt) [inline pkg/controller/bookkeepercluster/bookkeepercluster_controller.go:22]: Nit: Can we remove the extra newline

File: pkg/controller/bookkeepercluster/upgrade.go [hunk 2]
```diff
@@ -221,6 +221,12 @@ func (r *ReconcileBookkeeperCluster) syncBookkeeperVersion(bk *bookkeeperv1alpha
 
 		configMap := MakeBookieConfigMap(bk)
 		controllerutil.SetControllerReference(bk, configMap, r.scheme)
+		currentConfigMap := &corev1.ConfigMap{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: util.ConfigMapNameForBookie(bk.Name), Namespace: bk.Namespace}, currentConfigMap)
+		if err != nil {
+			return false, fmt.Errorf("failed to get configmap %v", err)
```
Comment (nishant-yt) [inline pkg/controller/bookkeepercluster/upgrade.go:227]:
```suggestion
			return false, fmt.Errorf("failed to get configmap (%s): %v", configMap,  err)
```

File: pkg/controller/bookkeepercluster/bookkeepercluster_controller_test.go [hunk 3]
```diff
@@ -13,10 +13,11 @@ package bookkeepercluster
 import (
 	"context"
 	"fmt"
-	"github.com/pravega/bookkeeper-operator/pkg/controller/config"
 	"strings"
 	"testing"
 
+	"github.com/pravega/bookkeeper-operator/pkg/controller/config"
+
```
Comment (nishant-yt) [inline pkg/controller/bookkeepercluster/bookkeepercluster_controller_test.go:20]: Nit: Can we remove the extra newline

File: pkg/controller/bookkeepercluster/upgrade.go (hunk 2 above)
Comment (anishakj) [reply pkg/controller/bookkeepercluster/upgrade.go]: Done

File: pkg/controller/bookkeepercluster/bookkeepercluster_controller.go (hunk 1 above)
Comment (anishakj) [reply pkg/controller/bookkeepercluster/bookkeepercluster_controller.go]: Done

**Review by nishant-yt — APPROVED** [review]
LGTM

**Review by jkhalack — APPROVED** [review]

**Pull request merged**